### PR TITLE
Merge newly-rebased lib into main

### PR DESCRIPTION
**Pull request by komish**
Just three things:
- Rebased Lib onto Main
- Fixed a bug in our logging where we were inadvertently not emitting variables because we didn't use `sprintf` https://github.com/redhat-openshift-ecosystem/openshift-preflight/commit/f5852ef50be9ac4cd19b77fbdae19b6395a09e8f
- Updated the LIBRARY.md doc to comply with the new logging changes. https://github.com/redhat-openshift-ecosystem/openshift-preflight/commit/2899cb0cfa6f1237bc61044262a37c5073832bd9

I left the original **lib** branch intact so we can use it as a reference, if needed. The first 6 commits here should be aligned with it, barring merge conflicts (which were mostly go mod related)

### CODE REVIEW

**Comment by komish** [comment]
golangci-lint was upset about my import ordering so I fixed that and made it its own commit, simply because it touched 17 files :roll_eyes:

**Comment by komish** [comment]
Re E2E Failures: Prow seems to be failing because some logic is creating a directory `artifacts` at the root of the source, and we already have that directory as a library.

**Review by sebrandon1 — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -31,8 +31,6 @@ results-junit.xml
 # E2E customzied env log file.
 sgol.txt
 
-# Testing dirs
-/artifacts
 # E2E customized env artifacts dir
 stcafitra/
```
Comment (skattoju) [inline .gitignore:35]: 🤨

**Review by acornett21 — APPROVED** [review]

**Review by skattoju — APPROVED** [review]
/lgtm

**Comment by komish** [comment]
:partying_face:

**Comment by sebrandon1** [comment]
Great to see this merged into main!  Thanks!

**Pull request merged**